File: Lab8_2/Lab8_2/ViewController.m
//
//  ViewController.m
//  Lab8_2
//
//  Created by robert on 5/28/17.
//  Copyright (c) 2017 di. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIView *regView;
@property (weak, nonatomic) IBOutlet UITextField *login;
@property (weak, nonatomic) IBOutlet UITextField *mail;
@property (weak, nonatomic) IBOutlet UITextField *pass;
@property (weak, nonatomic) IBOutlet UITextField *repPass;
@property (weak, nonatomic) IBOutlet UISwitch *accept;
@property (weak, nonatomic) IBOutlet UITextField *enterLogin;
@property (weak, nonatomic) IBOutlet UITextField *enterPass;
@property (weak, nonatomic) IBOutlet UIView *enterView;

@end

@implementation ViewController
- (IBAction)signUp:(id)sender {
    if (![_login.text isEqualToString:@""] && ![_pass.text isEqualToString:@""] && ![_mail.text isEqualToString:@""] && [_pass.text isEqualToString:_repPass.text] && [_accept isOn])
    {
        NSUserDefaults *user = [NSUserDefaults standardUserDefaults];
        [user setObject:_login.text forKey:@"login"];
        [user setObject:_pass.text forKey:@"pass"];
        //[self performSegueWithIdentifier:@"ClosedZone" sender:self];
    }

}
- (IBAction)enter:(id)sender {
    NSUserDefaults *user = [NSUserDefaults standardUserDefaults];
    if ([_enterLogin.text isEqualToString:(NSString*)[user objectForKey:@"login"]] && [_enterPass.text isEqualToString:(NSString*)[user objectForKey:@"pass"]])
    {
        [self performSegueWithIdentifier:@"ClosedZone" sender:self];
    }

}
- (IBAction)segment:(id)sender {
    UISegmentedControl *seg = (UISegmentedControl*)sender;
    if (seg.selectedSegmentIndex == 0)
    {
        _enterView.hidden = YES;
    }
    else
    {
        _enterView.hidden = NO;
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    _enterView.hidden = YES;
    NSUserDefaults *user = [NSUserDefaults standardUserDefaults];
    if ([user objectForKey:@"login"] != nil && [user objectForKey:@"pass"] != nil)
    {
        [self performSegueWithIdentifier:@"ClosedZone" sender:self];
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Lab8_3/Lab8_3/ViewController.swift
//
//  ViewController.swift
//  Lab8_3
//
//  Created by robert on 5/19/17.
//  Copyright (c) 2017 di. All rights reserved.
//

import UIKit
import CoreData

class ViewController: UIViewController, UITableViewDataSource {

    @IBOutlet weak var studentNameTextField: UITextField!
   
    @IBOutlet weak var table: UITableView!
    
    var students: [NSManagedObject] = []
    
    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("studentCell")! as UITableViewCell
        let student = students[indexPath.row]
        cell.textLabel?.text = student.valueForKey("name") as? String //Заполняем текст ячейки таблицы значением ключа "name"
        return cell
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection: Int) -> Int {
        return students.count
    }
    
    func tableView(tableView: UITableView, commitEditingStyle editingStyle: UITableViewCellEditingStyle, forRowAtIndexPath indexPath: NSIndexPath) {
        if editingStyle == UITableViewCellEditingStyle.Delete
        {
            let appDelegate = UIApplication.sharedApplication().delegate as AppDelegate //Создаем ссылку на класс AppDelegate из файла AppDelegate.swift
            let managedObjectContext = appDelegate.managedObjectContext //Создаем ссылку на метод managedObjectContext из класса AppDelegate в файле AppDelegate.swift
            managedObjectContext!.deleteObject(students[indexPath.row] as NSManagedObject) //Выбираем метод удаления объекта из модели данных students

            var error: NSError? = nil
            managedObjectContext?.save(&error)
            if (error == nil)
            {
                students.removeAtIndex(indexPath.row) //Удаляем объект из модели students
                tableView.deleteRowsAtIndexPaths([indexPath], withRowAnimation: .Left) //Удаляем строку из таблицы
            }
            else
            {
                print("Data removing error: \(error)") //В случае возникновения ошибки, выводим ее в консоль
            }
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func addStudentButton(sender: AnyObject) {
        if studentNameTextField.text == "" || studentNameTextField.text == "Введите данные!"
        {
            studentNameTextField.text = "Введите данные!"
        }
        else
        {
            let appDelegate = UIApplication.sharedApplication().delegate as AppDelegate //Создаем ссылку на класс AppDelegate из файла AppDelegate.swift
            let managedObjectContext = appDelegate.managedObjectContext //Создаем ссылку на объект managedObjectContext из класса AppDelegate
            let newObject = NSEntityDescription.insertNewObjectForEntityForName("Student", inManagedObjectContext:managedObjectContext!) as NSManagedObject //Создаем переменную, которая будет заносить новый объект в сущность "Students"
            newObject.setValue(studentNameTextField.text!, forKey: "name") //Значением нового объекта для ключа "name" будет являться текст из текстового поля studentNameTextField

            var error: NSError? = nil
            managedObjectContext?.save(&error)
            if (error == nil)
            {
                students.append(newObject) //Добавляем новый объект в модель данных students
                studentNameTextField.text! = "" //Очищаем текстовое поле
                self.table.reloadData() //Обновляем содержимое таблицы
                self.view.endEditing(true) //Убираем клавиатуру с экрана и выходим из режима редактировани
            }
            else
            {
                print("Data saving error: \(error)") //В случае возникновения ошибки, выводим ее в консоль
            }
        }
    }
    
    override func viewWillAppear(animated: Bool)
    {
        super.viewWillAppear(animated)
        let appDelegate = UIApplication.sharedApplication().delegate as AppDelegate //Создаем ссылку на класс AppDelegate из файла AppDelegate.swift
        let managedObjectContext = appDelegate.managedObjectContext //Создаем ссылку на объект managedObjectContext из класса AppDelegate
        let fetchRequest = NSFetchRequest(entityName: "Student") //Создаем запрос из сущности "Students"
        
        var error : NSError? = nil
        students = managedObjectContext?.executeFetchRequest(fetchRequest, error: &error) as [NSManagedObject]
        if (error != nil)
        {
            print("Data loading error: \(error)")
        }
        self.table.reloadData() //Обновляем содержимое таблицы
    }

}

